### PR TITLE
Enable `load_lora_weights` for `StableDiffusion3InpaintPipeline`

### PR DESCRIPTION
**Pull request by baldesco**
Enable `StableDiffusion3InpaintPipeline` to load lora weights, by using `SD3LoraLoaderMixin`. 

This emulate the way in which `StableDiffusion3ControlNetInpaintingPipeline` does it.

### CODE REVIEW

**Comment by baldesco** [comment]
@yiyixuxu @sayakpaul @asomoza

File: src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3_inpaint.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def retrieve_timesteps(
     return timesteps, num_inference_steps
 
 
-class StableDiffusion3InpaintPipeline(DiffusionPipeline):
+class StableDiffusion3InpaintPipeline(DiffusionPipeline, SD3LoraLoaderMixin, FromSingleFileMixin):
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3_inpaint.py:151]: Do we have a single-file checkpoint for inpainting yet? 

Additionally, could you provide an example showing the usage of a LoRA in the context of SD3 inpainting?

Comment (baldesco) [reply src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3_inpaint.py:151]: Here is a snippet showing an example of how to use a LoRA with SD3 inpainting

```
import torch
from diffusers.pipelines import StableDiffusion3InpaintPipeline
from diffusers.utils import load_image, make_image_grid
from dotenv import load_dotenv

load_dotenv('.env.local')

image = load_image(
    "https://huggingface.co/alimama-creative/SD3-Controlnet-Inpainting/resolve/main/images/dog.png"
)
mask = load_image(
    "https://huggingface.co/alimama-creative/SD3-Controlnet-Inpainting/resolve/main/images/dog_mask.png"
)

model_id = 'stabilityai/stable-diffusion-3-medium-diffusers'
adapter_id = 'zwloong/sd3-lora-training-rank32-v2'

pipe = StableDiffusion3InpaintPipeline.from_pretrained(
    model_id,
    torch_dtype=torch.float16,
)
pipe.to("cuda")

pipe.load_lora_weights(adapter_id)

output = pipe(
    prompt="a design of a cute pokemon featuring a bird",
    image=image,
    mask_image=mask,
    num_inference_steps=50,
    guidance_scale=7.5,
).images[0]

make_image_grid([image, mask, output], rows=1, cols=3)
```

result:

image

Comment (sayakpaul) [reply src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3_inpaint.py:151]: Ah nice!

**Comment by sayakpaul** [comment]
@yiyixuxu WDYT?

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
@sayakpaul @baldesco  looking great! merged

**Pull request merged**